src/Compute/Compute/Extension/Diagnostics/AddAzureRmVmssDiagnosticsExtension.cs

```diff
@@ -166,7 +166,7 @@ protected override void ProcessRecord()
 
                     newDiagnosticsExtension.Name = this.Name;
                     newDiagnosticsExtension.Publisher = DiagnosticsExtensionConstants.ExtensionPublisher;
-                    newDiagnosticsExtension.Type = DiagnosticsExtensionConstants.ExtensionType;
```

Comment:
1. Change the output of `Set-AzVmssDiskEncryptionExtension` may cause breaking change. I think it's a better way to make the fields of `VirtualMachineScaleSetExtension` stable.
1. `Type1` is not very clear, could you please change another name for it?

Reply:
This is because of the recent Swagger and SDK update.  Basically, the read-only "type" property was missing for extension for a long time, and the below commit fixed the issue.  However, "extension" has "type" property inside of "properties" (https://github.com/Azure/azure-rest-api-specs/blob/e63401cb21aba92650b877be0b7d8893452cd6fe/specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json#L6401) 
Then, after below PR is merged, SDK was generated so that the "type" inside of "properties" was changed from "type" to "type1", and the missing read-only "type" became "type" in the generated SDK by AutoRest.


https://github.com/Azure/azure-rest-api-specs/commit/0a7f5e5079a81337de3995e71b6e5ee4dbf48756#diff-be05d698662089e7c89933251de79202

Reply:
@wyunchi-ms   Also, note that the output of Set-AzVmssDiskEncryptionExtension does not include type.

https://github.com/Azure/azure-powershell/blob/master/src/Compute/Compute/Models/PSVmssVMDiskEncryptionStatusContext.cs

Reply:
> This is because of the recent Swagger and SDK update. Basically, the read-only "type" property was missing for vmss extension for a long time, and the below commit fixed the issue. However, "vmss extension" also has "type" property inside of "properties" (https://github.com/Azure/azure-rest-api-specs/blob/e63401cb21aba92650b877be0b7d8893452cd6fe/specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json#L9041)
> Then, after below PR is merged, SDK was generated so that the "type" inside of "properties" was changed from "type" to "type1", and the missing read-only "type" became "type" in the generated SDK by AutoRest.
> 
> [Azure/azure-rest-api-specs@0a7f5e5#diff-be05d698662089e7c89933251de79202](https://github.com/Azure/azure-rest-api-specs/commit/0a7f5e5079a81337de3995e71b6e5ee4dbf48756#diff-be05d698662089e7c89933251de79202)

But `Add-AzVmssDiagnosticsExtension` has wrote out `VirtualMachineScaleSetExtension` as the output.

Reply:
Sorry for forgeting to notice @hyonholee